homeassistant/components/fritz/switch.py

```diff
@@ -0,0 +1,522 @@
+"""Switches for AVM Fritz!Box functions."""
+from __future__ import annotations
+
+from collections import OrderedDict
+import logging
+from typing import Any
+
+from fritzconnection.core.exceptions import FritzConnectionException, FritzSecurityError
+import xmltodict
+
+from homeassistant.components.network.util import async_get_source_ip
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .common import FritzBoxBaseSwitch, FritzBoxTools, SwitchInfo
+from .const import (
+    DOMAIN,
+    SWITCH_TYPE_DEFLECTION,
+    SWITCH_TYPE_DEVICEPROFILE,
+    SWITCH_TYPE_PORTFORWARD,
+    SWITCH_TYPE_WIFINETWORK,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def service_call_action(
+    fritzbox_tools: FritzBoxTools,
+    service_name: str,
+    service_suffix: str | None,
+    action_name: str,
+    **kwargs: Any,
+) -> None | dict:
+    """Return service details."""
+
+    if f"{service_name}{service_suffix}" not in fritzbox_tools.connection.services:
+        return None
+
+    try:
+        return await fritzbox_tools.hass.async_add_executor_job(
+            lambda: fritzbox_tools.connection.call_action(
+                f"{service_name}:{service_suffix}", action_name, **kwargs
+            )
+        )
```

Comment:
```suggestion
        return await fritzbox_tools.hass.async_add_executor_job(
            partial(fritzbox_tools.connection.call_action,
                f"{service_name}:{service_suffix}", action_name, **kwargs
            )
        )
```

I guess @bdraco means something like this? (following up on your question on discord)

Do not forget to import functools:
`from functools import partial`

Reply:
Thx